src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -350,17 +360,27 @@ public DarCollectionSummary getSummaryForRoleNameByCollectionId(User user, Strin
           processDarCollectionSummariesForSO(List.of(summary));
           break;
         case CHAIRPERSON:
-          datasetIds = datasetDAO.findDatasetDTOsByUserId(userId).stream()
-              .map(d -> d.getDataSetId())
-              .collect(Collectors.toList());
+          List<Integer> chairDacIds = user.getRoles().stream()
+              .filter(ur -> ur.getRoleId().equals(UserRoles.CHAIRPERSON.getRoleId()))
+              .map(UserRole::getDacId)
+              .filter(Objects::nonNull)
+              .toList();
+          datasetIds = chairDacIds.isEmpty() ? List.of() : datasetDAO.findDatasetListByDacIds(chairDacIds).stream()
+              .map(Dataset::getDataSetId)
+              .toList();
           summary = darCollectionSummaryDAO.getDarCollectionSummaryForDACByCollectionId(userId,
               datasetIds, collectionId);
           processDarCollectionSummariesForChair(List.of(summary));
           break;
         case MEMBER:
-          datasetIds = datasetDAO.findDatasetDTOsByUserId(userId).stream()
-              .map(d -> d.getDataSetId())
-              .collect(Collectors.toList());
+          List<Integer> memberDacIds = user.getRoles().stream()
+              .filter(ur -> ur.getRoleId().equals(UserRoles.MEMBER.getRoleId()))
+              .map(UserRole::getDacId)
+              .filter(Objects::nonNull)
+              .toList();
+          datasetIds = memberDacIds.isEmpty() ? List.of() : datasetDAO.findDatasetListByDacIds(memberDacIds).stream()
```

Comment:
It looks like this block is duplicated four times, the only difference is the `UserRoles` value. Could it be refactored into a common method?
```
  private List<Integer> getDatasetIdsForUser(User user, UserRoles role) {
    List<Integer> chairDacIds = user.getRoles().stream()
        .filter(ur -> ur.getRoleId().equals(role.getRoleId()))
        .map(UserRole::getDacId)
        .filter(Objects::nonNull)
        .toList();
    return chairDacIds.isEmpty() ? List.of() : datasetDAO.findDatasetListByDacIds(chairDacIds).stream()
            .map(Dataset::getDataSetId)
            .toList();
  }
```

Reply:
Yes, this can be pulled out, thank you!